pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/ResponseStoreResource.java

```diff
@@ -0,0 +1,190 @@
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.container.Suspended;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.apache.pinot.common.cursors.AbstractResponseStore;
+import org.apache.pinot.common.metrics.BrokerMeter;
+import org.apache.pinot.common.metrics.BrokerMetrics;
+import org.apache.pinot.common.response.BrokerResponse;
+import org.apache.pinot.common.response.CursorResponse;
+import org.apache.pinot.core.auth.Actions;
+import org.apache.pinot.core.auth.Authorize;
+import org.apache.pinot.core.auth.ManualAuthorization;
+import org.apache.pinot.core.auth.TargetType;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.apache.pinot.spi.utils.CommonConstants;
+import org.glassfish.jersey.server.ManagedAsync;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.pinot.spi.utils.CommonConstants.SWAGGER_AUTHORIZATION_KEY;
+
+
+/**
+ *
+ */
+@Api(tags = "ResponseStore", authorizations = {@Authorization(value = SWAGGER_AUTHORIZATION_KEY)})
+@SwaggerDefinition(securityDefinition = @SecurityDefinition(apiKeyAuthDefinitions = @ApiKeyAuthDefinition(name =
+    HttpHeaders.AUTHORIZATION, in = ApiKeyAuthDefinition.ApiKeyLocation.HEADER, key = SWAGGER_AUTHORIZATION_KEY,
+    description = "The format of the key is  ```\"Basic <token>\" or \"Bearer <token>\"```")))
+@Path("/responseStore")
+public class ResponseStoreResource {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ResponseStoreResource.class);
+
+  @Inject
+  private PinotConfiguration _brokerConf;
+
+  @Inject
+  private BrokerMetrics _brokerMetrics;
+
+  @Inject
+  private AbstractResponseStore _responseStore;
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/")
+  @Authorize(targetType = TargetType.CLUSTER, action = Actions.Cluster.GET_RESULT_STORE)
+  @ApiOperation(value = "Get requestIds of all responses in the result store.", notes = "Get requestIds of all "
+      + "query stores in the result store")
+  public Collection<CursorResponse> getResults(@Context HttpHeaders headers) {
+    try {
+      return _responseStore.getAllStoredResponses();
+    } catch (Exception e) {
+      throw new WebApplicationException(e,
+          Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(e.getMessage()).build());
+    }
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("{requestId}")
+  @ApiOperation(value = "Response without ResultTable for a requestId")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Query response"), @ApiResponse(code = 500, message = "Internal Server Error")
+  })
+  @ManualAuthorization
+  public BrokerResponse getSqlQueryMetadata(
+      @ApiParam(value = "Request ID of the query", required = true) @PathParam("requestId") String requestId) {
```

Comment:
Does this `@ManualAuthorization` means that anyone can get the metadata of any known request id?

Reply:
I've messed up my git branches. I have the code in a branch. I'll merge the code.